redis/asyncio/sentinel.py

```diff
@@ -167,6 +193,92 @@ async def rotate_slaves(self) -> AsyncIterator:
+        connection = None
+        # If this is the first scan request of the iter command,
+        # get a connection from the pool
+        if server_host is None or server_port is None:
+            try:
+                connection = self._available_connections.pop()
+            except IndexError:
+                connection = self.make_connection()
+        # If this is not the first scan request of the iter command
+        else:
+            # Check from the available connections, if any of the connection
+            # is connected to the host and port that we want
+            for available_connection in self._available_connections.copy():
+                # if yes, use that connection
+                if (
+                    available_connection.host == server_host
+                    and available_connection.port == server_port
+                ):
+                    self._available_connections.remove(available_connection)
+                    connection = available_connection
+            # If not, make a new dummy connection object, and set its host and port
+            # to the one that we want later in the call to ``connect_to_address``
+            if not connection:
+                connection = self.make_connection()
+        assert connection
+        self._in_use_connections.add(connection)
+        try:
+            # Ensure this connection is connected to Redis
+            # If this is the first scan request, it will
+            # call rotate_slaves and connect to a random replica
+            if server_port is None or server_port is None:
+                await connection.connect()
+            # If this is not the first scan request,
+            # connect to the previous replica.
+            # This will connect to the host and port of the replica
+            else:
+                await connection.connect_to_address(server_host, server_port)
+            # Connections that the pool provides should be ready to send
+            # a command. If not, the connection was either returned to the
+            # pool before all data has been read or the socket has been
+            # closed. Either way, reconnect and verify everything is good.
+            try:
+                if await connection.can_read_destructive():
+                    raise ConnectionError("Connection has data") from None
+            except (ConnectionError, OSError):
+                await connection.disconnect()
+                await connection.connect()
+                if await connection.can_read_destructive():
+                    raise ConnectionError("Connection not ready") from None
+        except BaseException:
+            # Release the connection back to the pool so that we don't
+            # leak it
+            await self.release(connection)
+            raise
+        # Store the connection to the dictionary
+        self._request_id_to_replica_address[iter_req_id] = (
```

Comment:
When would these entries be removed from the dict? So that it does not grow indefinitely.

Reply:
Yeah good point, I created a separate `cleanup` method that will be called at the end of scan iter family commands